Authentication/JWT.php

```
@@ -134,6 +179,34 @@ public static function encode($payload, $key, $alg = 'HS256', $keyId = null, $he
         if ( isset($head) && is_array($head) ) {
             $header = array_merge($head, $header);
         }
+        if (isset($options['audience']) && (is_string($options['audience']) || is_array($options['audience']))) {
+            if (is_array($payload)) {
+                $payload['aud'] = is_array($options['audience']) ? $options['audience'] : array($options['audience']);
+            } else if (is_object($payload)) {
+                $payload->aud = is_array($options['audience']) ? $options['audience'] : array($options['audience']);
+            }
```

The `else` code path for this field, and the three below it, should probably throw an exception if the type is invalid rather than silently fail to be added to the token.